### PR TITLE
Update clean-css to v3.x.

### PR DESCRIPTION
**Pull request by XhmikosR**
/CC @kangax @duncanbeevers in case you sort this faster.

TODO
- [ ] Fix tests!!!

### CODE REVIEW

**Comment by XhmikosR** [comment]
/CC @jakubpawlowicz: any idea why tests keep failing?

**Comment by jakubpawlowicz** [comment]
I'm not familiar with this code but can check in the evening. Unless someone fixes it sooner!

**Comment by silverwind** [comment]
The test failure comes down to inline styles not being minified, probably because the style attribute string likely isn't treated as valid CSS in itself.

@jakubpawlowicz should style contents without a selector still be supported? Here's the nonworking test:

``` js
cleanCSS.minify('color: red; background-color: yellow; font-family: Verdana, Arial, sans-serif').styles
// -> 'color: red; background-color: yellow; font-family: Verdana, Arial, sans-serif'
```

**Comment by XhmikosR** [comment]
Rebased and updated to 3.0.1.

**Comment by silverwind** [comment]
Wrapping the style string in a `* {}` works as expected, by the way:

``` js
new (require("clean-css"))().minify("color: red; background-color: yellow; font-family: Verdana, Arial, sans-serif").styles
// -> 'color: red; background-color: yellow; font-family: Verdana, Arial, sans-serif'
new (require("clean-css"))().minify("*{color: red; background-color: yellow; font-family: Verdana, Arial, sans-serif}").styles
// ->  '*{color:red;background-color:#ff0;font-family:Verdana,Arial,sans-serif}'

```

**Comment by jakubpawlowicz** [comment]
@silverwind as mentioned in https://github.com/jakubpawlowicz/clean-css/issues/418 such content is deemed invalid now, but :100: for wrapping it in a dummy selector.

**Comment by silverwind** [comment]
I guess we'd need to wrap, minfiy and unwrap the contents of an inline style now. I could do a PR, unless anyone else has a better idea :)

**Comment by XhmikosR** [comment]
@silverwind: sure, but use this branch as a base for your PR and CC me when you are done.

**Comment by silverwind** [comment]
@XhmikosR #317 done

**Comment by XhmikosR** [comment]
@silverwind: Awesome, thanks! I tested the changes locally and work fine.

I'll merge this in a few minutes.

File: src/htmlminifier.js [hunk 1]
```diff
@@ -266,6 +266,22 @@
...
   }
 
+  // Wrap CSS declarations for CleanCSS > 3.x
+  // See https://github.com/jakubpawlowicz/clean-css/issues/418
+  function wrapCSS(text) {
+    return '*{' + text + '}';
+  }
+
+  function unwrapCSS(text) {
+    var matches = text.match(/^\*\{((.*|\n)*)\}$/m);
```
Comment (duncanbeevers) [inline src/htmlminifier.js:276]: This regexp seems overly-complicated. Can it be replaced like so?

``` javascript
var matches = text.match(/^\*\{([\s\S]*)\}$/m);
```

regexps

This uses fewer capture groups, and accepts all characters.

Comment (XhmikosR) [inline src/htmlminifier.js:276]: You can always replace it yourself if you think it's better and make a PR.

Comment (silverwind) [inline src/htmlminifier.js:276]: Yeah, yours looks cleaner with one less capture group, and seems to work fine:

http://regexr.com/3a4d1
http://regexr.com/3a4cu

Comment (silverwind) [inline src/htmlminifier.js:276]: Actually just noticed a small mistake in mine, the first `*` inside the capture group is unnecessary. So please do a PR :)

Comment (XhmikosR) [inline src/htmlminifier.js:276]: Or actually, just push it since @duncanbeevers has push rights (I forgot about it) :p

**Pull request merged**